learn/advanced/pagination.md

```diff
@@ -0,0 +1,117 @@
+
+Instead of a full page selector, [we recommend creating pagination interfaces centered around previous and next buttons](#recommended-previous-and-next-buttons).
+
+::: note
+By default, Meilisearch returns a maximum of 1000 search results. Consult our [index setting reference](/reference/api/settings.md) if you need to change this.
+:::
+
+## Recommended: Previous and next buttons
+
+Pagination interfaces that rely solely on previous and next buttons are similar to page selectors, but do not allow users to jump to an arbitrary results page.
+
+![Placeholder image for prev/next interface]()
+
+Though this approach offers less precision than a full-blown page selector, it does not require knowing the number of total search results.
+
+### Implementation
+
+Previous and next buttons can be implemented using the [`limit`](/reference/api/search.md#limit) and [`offset`](/reference/api/search.md#offset) search parameters.
+
+Paginating requires providing an `offset` that is equal to your `limit` times the page number times: `offset = limit * page number`. We recommend starting page count from 0 instead of 1 when calculating offset values.
+
+For example, if you set `limit` to `20` and want the first page of search results, your `offset` must be 0: `offset = 0 * 20`.
+
+```js
+const results = await index.search("x", { limit: 20, offset: 0 });
+```
+
+If you keep the same `limit` and want the third page of results, `offset` must be `40`: `offset = 2 * 20`.
+
+```js
+const results = await index.search("x", { limit: 20, offset: 40 });
+```
+
+Once a query returns less than your configured `limit`, you have reached the last results page.
+
+It is often helpful to disable navigation buttons when the user cannot move to the "next" or "previous" page. The "previous" button should be disabled whenever your `offset` is 0.
+
+To disable the "next" button, we recommend setting your query's `limit` to the number of displayed results plus one. The extra result indicates that you have at least one item beyond what you can display in a single results page:
+
+```js
+const results = await index.search("x", { limit: 21, offset: 0 })
+
+if (offset === 0 ) {
+  // If offset equals 0, we're on the first results page
+  // and can disable the "previous" buttons
+  document.querySelector("#previous_button").disabled = true;
+} else {
+  document.querySelector("#previous_button").disabled = false;
+}
+
+if (results.hits.length < 21 ) {
+  // If Meilisearch returns 20 items or less, 
+  // we don't have any more results to 
+  // and can disable the "next"
+  document.querySelector("#next_button").disabled = true;
+} else {
+  // If Meilisearch returns exactly 21 results
+  // and our page can only show 20 items at a time,
+  // we have at least one more page with 1 result in it
```

Comment:
I think this comment isn't necessary. To me it seems self explanatory, but I could be wrong 🤔